src/utils.jl

```diff
@@ -458,16 +458,38 @@ function flatten1(ex)
   for x in ex.args
     isexpr(x, :block) ? append!(ex′.args, x.args) : push!(ex′.args, x)
   end
-  # Don't use `unblock` to preserve line nos
+  # Don't use `unblock` to preserve line nos.
   return length(ex′.args) == 1 ? ex′.args[1] : ex′
 end
 
+# Helpers for flattening try blocks
+false_or_symbol_or_bflatten(x) = x==false || isa(x, Symbol) ? x : bflatten(x)
+function bflatten(x) # flatten down to a block (i.e. a single symbol is turned into `begin symbol end`, unlike `flatten` which would just return the symbol)
+    fx = flatten(x)
+    return isexpr(fx, :block) ? fx : Expr(:block,fx)
+end
+
 """
     flatten(ex)
 
 Flatten any redundant blocks into a single block, over the whole expression.
 """
-flatten(ex) = postwalk(flatten1, ex)
+function flatten end
+
+flatten(x) = x
+function flatten(x::Expr)
+  if isexpr(x, :try) # try args can be either false or blocks, but not non-block exprs, except for the second argument that has to be false or symbol... so we need to be a bit verbose here
```

Comment:
That comment is only true for the parser. Macro code can form expressions like `eval(Expr(:try, 1, false, 3))`, and julia takes it just fine. I don't think we should trigger a new error on an `Expr` like that.

Reply:
I will clarify the comment. I do not think we raise an error on your example, but we do modify it. I will add tests.